### DISCUSSION TITLE
Online PIOASM IS Available

### DISCUSSION THREAD

**Discussion by jake653** [earlephilhower/arduino-pico#193] [Show and tell]
Online pioasm for Raspberry Pi Pico, Assembler, verified .pio.h output identical to servo example

The Tool https://wokwi.com/tools/pioasm
Github src  (Posted Jun. 3) https://github.com/wokwi/pioasm-wasm

There is also Docker code available
https://github.com/kahara/pioasm-docker

- - - - 
With my limited time and ancient hardware with older sys. , I have not always succeeded installing tool chains.
 Looks like I can now use the cheap Pi Pico without the bother...

**Comment by earlephilhower** [comment]
Very cool! and ingenious use of WASM.

FWIW, the repo installation also has a precompiled `pioasm` for CLI use, but you need to manually find where the Arduino IDE placed it (~/.arduino15/....somewhere.../pioasm/pioasm).

**Reply by earlephilhower** [reply]
I just added a link in the docs to your tool.  Hopefully that will make it easier to find for folks!
